### PR TITLE
Add support to add additional configuration to opensearch-dashboards.yml

### LINKED ISSUE #72: [FEATURE] Ability to pass custom configuration to OpenSearch-Dashboards.yml file
### Is your feature request related to a problem?
We have the feature in add customized cluster settings by adding config to opensearch.yml https://github.com/opensearch-project/opensearch-cluster-cdk/pull/21
However, this feature is lacking for opensearch-dashboards

### What solution would you like?
Ability to pass custom configuration to OpenSearch-Dashboards.yml file

### What alternatives have you considered?
_A clear and concise description of any alternative solutions or features you've considered._

### Do you have any additional context?
_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Add support to add additional configuration to opensearch-dashboards.yml

### Issues Resolved
resolves #72 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
 | jvmSysProps            | Optional    | string  | A comma-separated list of key=value pairs that will be added to `jvm.options` as JVM system properties.                                                                                                                                                                                          |
 | additionalConfig       | Optional    | string  | Additional opensearch.yml config parameters passed as JSON. e.g., `--context additionalConfig='{"plugins.security.nodes_dn": ["CN=*.example.com, OU=SSL, O=Test, L=Test, C=DE", "CN=node.other.com, OU=SSL, O=Test, L=Test, C=DE"], "plugins.security.nodes_dn_dynamic_config_enabled": false}'` |
+| additionalOSDconfig       | Optional    | string  | Additional opensearch_dashboards.yml config parameters passed as JSON. e.g., `additionalOSDconfig='{"data.search.usageTelemetry.enabled": "true"}'` |
```
Comment (rishabh6788) [inline README.md:59]: Nit: maybe `additionalOsdconfig` to conform with camel-casing?

File: lib/infra/infra-stack.ts [hunk 2]
```diff
@@ -611,20 +612,30 @@ export class InfraStack extends Stack {
...
       }
 
       cfnInitConfig.push(InitCommand.shellCommand('set -ex;cd opensearch-dashboards;'
-          + 'sudo -u ec2-user nohup ./bin/opensearch-dashboards > dashboard_install.log 2>&1 &', {
+        + 'sudo -u ec2-user nohup ./bin/opensearch-dashboards > dashboard_install.log 2>&1 &', {
         cwd: '/home/ec2-user',
         ignoreErrors: false,
       }));
+
+      // @ts-ignore
```
Comment (rishabh6788) [inline lib/infra/infra-stack.ts:630]: This setting should be added before dashboards are started else they will not be picked up.

File: lib/infra/infra-stack.ts [hunk 3]
```diff
@@ -611,20 +612,30 @@ export class InfraStack extends Stack {
...
+      if (props.additionalOSDconfig.toString() !== 'undefined') {
```
Comment (rishabh6788) [inline lib/infra/infra-stack.ts:631]: This setting should be added before dashboards are started else they will not be picked up.

File: README.md (hunk 1 above)
Comment (gaiksaya) [reply README.md]: Fixed everywhere!

File: lib/infra/infra-stack.ts (hunk 3 above)
Comment (gaiksaya) [reply lib/infra/infra-stack.ts]: Moved before the start! Thanks for catching this.

**Review by rishabh6788 — APPROVED** [review]
Thanks for adding this. LGTM!

**Comment by gaiksaya** [comment]
@rishabh6788 Do we want to backport this to 1.x too?

**Pull request merged**